resources/build/conanfile.py

```diff
@@ -0,0 +1,44 @@
+from conans import ConanFile, CMake
+
+
+class OpenAssetIOConan(ConanFile):
+    generators = (
+        # Generate a CMake toolchain preamble file `conan_paths.cmake`,
+        # which augments package search paths with conan package
+        # directories.
+        "cmake_paths",
+        # Generate `Find<PackageName>.cmake` finders, required for various
+        # public ConanCenter packages (e.g. pybind11) since they disallow
+        # bundling of `<PackageName>Config.cmake`-like files in the package.
+        "cmake_find_package",
+    )
+    settings = "os"
+    options = {"testing": [True, False], "install_cpython" : [True, False]}
+    # TODO(DF): For minimal friction, default install unit test
+    #  libraries for now - we should switch this off by default when
+    #  we get to distributing a conan package.
+    default_options = {"testing": True, "install_cpython" : True}
+
+    def build_requirements(self):
+        # CY2022
+        if self.options.install_cpython :
```

Comment:
[fix] I think for similar reasons as in https://github.com/OpenAssetIO/OpenAssetIO/pull/663#discussion_r1000539083 (i.e. avoiding affecting the package ID) we should use an environment variable rather than `options`.

We should probably also rename it to something other than install.

So an env var like `OPENASSETIO_CONAN_REQUIRE_CPYTHON` or similar.

Reply:
Addressed [48329fc](https://github.com/OpenAssetIO/OpenAssetIO/pull/665/commits/48329fc01c576948332fbb8ded6f4b4c32f7a379)

Worth giving this one a look, made some minor decisions. Called the var `OPENASSETIO_CONAN_SKIP_CPYTHON`, and removed a bit of redundant setting in the bootstrap files.

Reply:
Oh! and conan also had a [get_env](https://docs.conan.io/en/latest/reference/tools.html#tools-get-env) command which looks to be designed to solve that true/false/0/1 problem we were discussing.

Reply:
I like this name 👍

Reply:
Awesome. Actual commit seems to be c88ed675.  Much cleaner.

Reply:
I notice the latest [Ubuntu CI job](https://github.com/OpenAssetIO/OpenAssetIO/actions/runs/3313127655/jobs/5470684152) is installing `cpython` through conan, though.

Reply:
Actually, not quite all good - Windows builds are still installing cpython.

Reply:
Tricky though, since we need to set `CIBUILDWHEEL` in the `bootstrap_platform` steps.  Hopefully we can set a global env var for that workflow and it'll work?

Reply:
As we may have other situations where we want to use system python (Eg: pinning python version for ASAN, once we reinstate 3.7 support), would it make sense to just call that var `OPENASSETIO_CONAN_SKIP_CPYTHON`? and set it from the outside whenever we know we want to inherit python? Vs associating it solely with `cibuildwheel`?

Reply:
@feltech This what you were thinking? Looks to be not treating cpython as a dependency on the runners now : [8baa986](https://github.com/OpenAssetIO/OpenAssetIO/pull/665/commits/8baa9863ea437fd06688ae37268bb79fcab1a978)

(Went back and forth a bit after that fixup on the manylinux bootstrap, which maintains use of the `CIBUILDWHEEL` to setup itself rather than attempting to inject our environment variable through, which in retrospect is likely the entire point of that env variable being in the VM.)